cpp/src/arrow/engine/substrait/options.h

```diff
@@ -32,7 +36,7 @@ namespace engine {
 /// How strictly to adhere to the input structure when converting between Substrait and
 /// Acero representations of a plan. This allows the user to trade conversion accuracy
 /// for performance and lenience.
-enum class ConversionStrictness {
+enum class ARROW_ENGINE_EXPORT ConversionStrictness {
```

Comment:
What does the macro `ARROW_ENGINE_EXPORT` do?

Reply:
It's a [visibility macro](https://github.com/apache/arrow/pull/14385/files#diff-e803a80147d4db13f4a3eee8126ce922ca91a6c6c2295ca3c60d292514c66d1fR22). It tells the linker to export the symbols for the construct following it.